components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2390,16 +2390,38 @@ public String getSubscriber(String subscriptionId) throws APIManagementException
         Map<String, String> claimMap = new HashMap<>();
         try {
             tenantId = getTenantId(tenantDomain);
-        SortedMap<String, String> subscriberClaims =
-                APIUtil.getClaims(subscriber, tenantId, ClaimsRetriever.DEFAULT_DIALECT_URI);
-        APIManagerConfiguration configuration = getAPIManagerConfiguration();
-        String configuredClaims = configuration
-                .getFirstProperty(APIConstants.API_PUBLISHER_SUBSCRIBER_CLAIMS);
-        if (subscriberClaims != null) {
-            for (String claimURI : configuredClaims.split(",")) {
-                claimMap.put(claimURI, subscriberClaims.get(claimURI));
+            SortedMap<String, String> subscriberClaims =
+                    APIUtil.getClaims(subscriber, tenantId, ClaimsRetriever.DEFAULT_DIALECT_URI);
+            APIManagerConfiguration configuration = getAPIManagerConfiguration();
+            String configuredClaims = configuration
+                    .getFirstProperty(APIConstants.API_PUBLISHER_SUBSCRIBER_CLAIMS);
+            if (subscriberClaims != null) {
+                for (String claimURI : configuredClaims.split(",")) {
+                    claimMap.put(claimURI, subscriberClaims.get(claimURI));
+                }
             }
+        } catch (UserStoreException e) {
+            throw new APIManagementException("Error while retrieving tenant id for tenant domain "
+                    + tenantDomain, e);
         }
+        return claimMap;
+    }
+
+    /**
+     * Returns the claims of a User
+     *
+     * @param userName The name of the user
+     * @return The looked up claims of the user
+     * @throws APIManagementException if failed to get user
+     */
+    @Override
+    public Map<String, String> getLoggedInUserClaims(String userName) throws APIManagementException {
+        String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+        int tenantId = 0;
+        Map<String, String> claimMap;
+        try {
+            tenantId = getTenantId(tenantDomain);
+            claimMap = APIUtil.getClaims(userName, tenantId, ClaimsRetriever.DEFAULT_DIALECT_URI);
```

Comment:
I feel this method doesn't look right to have in APIProviderImpl.
Shall we move this method to APIUtil itself?

Reply:
Ideal way to is to have these types of method in a separate Util class.. like APIMUserMgtUtils. please check with others as well.

Reply:
In the APIUtil class, we have the base claims retriever method(getClaims()). The use cases which are using that method should come from the APIProviderImpl or APIConsumerImpl nah?

Reply:
It is not necessary. APIProvider and APIConsumer classes are actually APIM Domain specific..
APIProvider -> publisher operations
APIConsumer -> devportal operations
Retrieving user information is actually bit general and not specific to above two. So it is okay to have it in a Util class itself.

Reply:
According to our offline discussion created a new class **APIRealmUtil** to address this. Later we will port all user and role related methods from **APIUtil** class to this newly created class.

https://github.com/wso2/carbon-apimgt/pull/9454/commits/9dce063165691739762701864b94109afae195e2